### PR TITLE
CONSOLE-4073,CONSOLE-4074: Refactor Pull Secret forms

### PR DESCRIPTION
**Pull request by TheRealJon**
Refactor CreateConfigSubForm, PullSecretCredentialsForm, PullSecretCredentialEntry, and UploadConfigSubForm.

### CODE REVIEW

**Comment by TheRealJon** [comment]
QE Approver
/assign @yapei

Docs Approver:
/assign @opayne1

PX Approver:
/assign @reestr

**Comment by TheRealJon** [comment]
This PR contains the same changes as https://github.com/openshift/console/pull/14052, just from my branch to help expedite any further changes.

**Comment by yapei** [comment]
@XiyunZhao will work on PR testing and will report back once complete

**Comment by XiyunZhao** [comment]
This PR has been tested (regression and manual testing) without any issue, the refactor for pull secret forms works as expected
/label qe-approved

File: frontend/public/components/secrets/create-secret/GenericSecretForm.tsx [hunk 1]
```diff
@@ -7,11 +7,11 @@ import { Button } from '@patternfly/react-core';
 import { MinusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/minus-circle-icon';
 import { PlusCircleIcon } from '@patternfly/react-icons/dist/esm/icons/plus-circle-icon';
 import * as ITOB from 'istextorbinary/edition-es2017';
-import { KeyValueEntryFormState, SecretStringData } from './types';
 import { KeyValueEntryForm } from './KeyValueEntryForm';
+import { KeyValueEntry, SecretSubFormProps } from './types';
```
Comment (Mylanos) [inline frontend/public/components/secrets/create-secret/GenericSecretForm.tsx:11]: These changes in the `GenericSecretForm` are out of scope of the stories tackled in this PR, did you perhaps make those instinctively? I addressed these in [PR #14520](https://github.com/openshift/console/pull/14520/files), so keeping them might lead to some unnecessary conflicts. However, if you’d prefer to keep them, I can handle the conflicts during my rebase.

**Comment by Mylanos** [comment]
Just one comment, otherwise LGTM. @TheRealJon

File: frontend/public/components/secrets/create-secret/GenericSecretForm.tsx (hunk 1 above)
Comment (TheRealJon) [reply frontend/public/components/secrets/create-secret/GenericSecretForm.tsx:11]: These changes were necessary as part of some adjustments that were made to some shared type definitions.

**Comment by TheRealJon** [comment]
@Mylanos Do you mind reviewing again and tagging?

**Comment by Mylanos** [comment]
/LGTM
sorry for the delay

**Pull request merged**